src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp

```diff
@@ -0,0 +1,207 @@
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace CurvedScalarWave::Worldtube::Actions {
+/*!
+ * \brief Projects the regular field \f$\Psi^R\f$ and its time derivative
+ * \f$\partial_t \Psi^R\f$ onto real spherical harmonics and sends the result to
+ * the worldtube.
+ *
+ * \details The regular field is obtained by subtracting the singular/puncture
+ * field from the numerical DG field.
+ * All spherical harmonics are computed for \f$l <= n\f$, where \f$n\f$ is the
+ * worldtube expansion order. The projection is done by integrating over the DG
+ * grid of the element face using \ref definite_integral with the euclidean area
+ * element. The worldtube adds up all integrals from the different elements to
+ * obtain the integral over the entire sphere.
+ *
+ * DataBox:
+ * - Uses:
+ *    - `tags_to_slice_on_face`
+ *    - `Worldtube::Tags::ExpansionOrder`
+ *    - `Worldtube::Tags::FaceCoordinates<Dim, Frame::Grid, true>`
+ *    - `Worldtube::Tags::PunctureField`
+ *    - `Worldtube::Tags::ExcisionSphere`
+ *    - `Tags::TimeStepId`
+ */
+struct SendToWorldtube {
+  static constexpr size_t Dim = 3;
+  using tags_to_send = tmpl::list<CurvedScalarWave::Tags::Psi,
+                                  ::Tags::dt<CurvedScalarWave::Tags::Psi>>;
+
+  using tags_to_slice_to_face = tmpl::list<
+      CurvedScalarWave::Tags::Psi, CurvedScalarWave::Tags::Pi,
+      CurvedScalarWave::Tags::Phi<Dim>,
+      gr::Tags::Shift<Dim, Frame::Inertial, DataVector>,
+      gr::Tags::Lapse<DataVector>,
+      domain::Tags::InverseJacobian<Dim, Frame::ElementLogical, Frame::Grid>>;
+
+  using inbox_tags = tmpl::list<Worldtube::Tags::SphericalHarmonicsInbox<Dim>>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    const auto& puncture_field =
+        db::get<Worldtube::Tags::PunctureField<Dim>>(box);
+    if (puncture_field.has_value()) {
+      const auto& element_id = db::get<domain::Tags::Element<Dim>>(box).id();
+      const auto& excision_sphere = db::get<Tags::ExcisionSphere<Dim>>(box);
+      const auto direction = excision_sphere.abutting_direction(element_id);
+      ASSERT(direction.has_value(), "Should be abutting");
+      const auto& mesh = db::get<domain::Tags::Mesh<Dim>>(box);
+      const auto face_mesh = mesh.slice_away(direction->dimension());
+      const size_t face_size = face_mesh.number_of_grid_points();
+
+      Variables<tmpl::flatten<
+          tmpl::list<tags_to_slice_to_face, ::Tags::TempScalar<0>>>>
```

Comment:
[optional] You could also replace `flatten<...>` with `push_back<tags_to_slice_to_face, ::Tags::TempScalar<0>>`